packages/core/src/lib/lit-core.ts

```diff
@@ -634,17 +635,44 @@ export class LitCore {
     const connectedNodes = new Set<string>();
     const serverKeys: Record<string, JsonHandshakeResponse> = {};
 
-    let timeoutHandle: ReturnType<typeof setTimeout>;
-    await Promise.race([
-      new Promise((_resolve, reject) => {
-        timeoutHandle = setTimeout(() => {
+    // Handshake with each node
+    const handshakePromises = this.config.bootstrapUrls.map(async (url) => {
+      serverKeys[url] = await this._handshakeAndVerifyNodeAttestation({
+        url,
+        requestId,
+      });
+      connectedNodes.add(url);
+    });
+
+    // Succeed immediately when all nodes handshaked or at timeout when more than threshold handshaked
+    // Fail at threshold unreachable or below threshold at timeout
+    await new Promise<void>((resolve, reject) => {
+      let successes = 0;
+      let errors = 0;
+      const timeout = setTimeout(() => check(true), this.config.connectTimeout);
+
+      const check = (isTimeout: boolean) => {
+        // Force finish at timeout or when all nodes have handshaked/failed
+        const mustFinish =
```

Comment:
With settled it will resolve when all the promises have resolved, but we could reject before timeout when threshold is not reachable anymore

Reply:
Also, you will not have the allSettled promise resolved and its values at timeout if one of them is still pending so the operation would still need to be done over each `handshakePromise`

Reply:
I am not sure I agree on the first point as if the settled promise resolves it means all of the `handshake` promises have either `fufilled` or `rejected` but each will have to be in either state for the `settlingPrms` to resolve. However, with @MaximusHaximus Im not sure this matters because as he said we have to connect to al nodes as the `ECDSA` and `BLS` protocols could result in undefined behaviors if one of the nodes in the active set are not sent  a request during either
- BLS operations
- ECDSA operations

Reply:
Yeah, sorry about settled I mean either resolved or rejected. The goal of this PR is to not treat all nodes as one single block if we don't strictly have to, therefore allowing for 8 of 9 to be a successful handshake at timeout instead of failing for something you probably don't need or can get later

After clarifying this requirement this will be adjusted so I'm putting this as draft for now